package/MDAnalysis/coordinates/CHEMFILES.py

```diff
@@ -0,0 +1,240 @@
+        if frame.natoms() != self.n_atoms:
+            raise IOError(
+                "The number of atom changed in the trajectory. This is not " +
+                "supported by MDAnalysis."
+            )
+
+        cell = frame.cell()
+        ts.dimensions[:] = cell.lengths() + cell.angles()
+        ts.positions[:] = frame.positions()
+        if frame.has_velocities():
+            ts.has_velocities = True
+            ts.velocities[:] = frame.velocities()
+
+    @property
+    def dimensions(self):
+        """unitcell dimensions (*A*, *B*, *C*, *alpha*, *beta*, *gamma*)
+        """
+        return self.ts.dimensions
+
+    def Writer(self, filename, n_atoms=None, **kwargs):
+        """Return writer for trajectory format"""
+        if n_atoms is None:
+            n_atoms = self.n_atoms
+        return ChemfilesWriter(filename, n_atoms, **kwargs)
+
+
+class ChemfilesWriter(base.WriterBase):
+    """
+    Coordinate writer using chemfiles.
+
+    The file format to used is guessed based on the file extension. If no
+    matching format is found, a ``ChemfilesError`` is raised. It is also
+    possible to manually specify the format to use for a given file.
+    """
+
+    format = 'CHEMFILES'
+    multiframe = True
+    units = {'time': 'fs', 'length': 'Angstrom'}
+
+    def __init__(self, filename, n_atoms=0, mode="w", format="", **kwargs):
+        """Initialize the Chemfiles trajectory writer
+
+        Parameters
+        ----------
+        filename: str
+            filename of trajectory file.
+        n_atoms: int
+            number of atoms in the trajectory to write
+        mode : str (optional)
+            file opening mode: use 'a' to append to an existing file or 'w' to
+            create a new file
+        format : str (optional)
+            use the given format name instead of guessing from the extension.
+            The list of supported formats and their names is available in
+            chemfiles documentation, at
+            http://chemfiles.org/chemfiles/latest/formats.html
+        """
+        self.filename = filename
+        self.n_atoms = n_atoms
+        if mode != "a" and mode != "w":
+            raise IOError("Expected 'a' or 'w' as mode in ChemfilesWriter")
+        self._file = Trajectory(self.filename, mode, format)
+        self._closed = False
+
+    def close(self):
+        """Close the trajectory file and finalize the writing"""
+        if hasattr(self, "_closed") and not self._closed:
+            self._file.close()
+            self._closed = True
+
+    def write_next_timestep(self, ts):
+        """Write timestep object into trajectory.
+
+        Parameters
+        ----------
+        ts: TimeStep
+        """
+        frame = self._ts_to_frame(ts)
+        self._file.write(frame)
+
+
+    def _ts_to_frame(self, ts):
+        """
+        Convert a Timestep to a chemfiles Frame
+        """
+        if ts.n_atoms != self.n_atoms:
```

Comment:
What should I do in this case ? Chemfiles have no issue with varying number of atoms in a trajectory (except if the underlying format does not support it), so writing would work. But then it would not be possible to read it again with MDAnalysis.

Reply:
The writer can be free to vary n_atoms, it's only on the input side that we can't have it vary

Reply:
Ok, I'll document this behavior then.